src/inspector_agent.cc

```diff
@@ -424,67 +422,52 @@ bool AgentImpl::OnInspectorHandshakeIO(inspector_socket_t* socket,
   }
 }
 
-// static
-void AgentImpl::OnRemoteDataIO(uv_stream_t* stream,
-                           ssize_t read,
-                           const uv_buf_t* b) {
-  inspector_socket_t* socket = static_cast<inspector_socket_t*>(stream->data);
-  AgentImpl* agent = static_cast<AgentImpl*>(socket->data);
-  Mutex::ScopedLock scoped_lock(agent->pause_lock_);
+void AgentImpl::OnRemoteDataIO(inspector_socket_t* socket,
+                               ssize_t read,
+                               const uv_buf_t* buf) {
+  Mutex::ScopedLock scoped_lock(pause_lock_);
   if (read > 0) {
-    std::string str(b->base, read);
-    agent->PushPendingMessage(&agent->message_queue_, str);
-    free(b->base);
-
+    String16 str = String16::fromUTF8(buf->base, read);
+    free(buf->base);
```

Comment:
Inspector socket passes nullptr instead of buf on error. Not sure how compatible this is with libuv conventions - I moved the free to always dispose non-null buffer.